lib/archethic_web/controllers/api/web_hosting_controller.ex

```diff
@@ -69,6 +64,29 @@ defmodule ArchethicWeb.API.WebHostingController do
     end
   end
 
+  defp encode_res(conn, file_content, encodage) do
+    accept_encoding = get_req_header(conn, "accept-encoding")
+
+    case !Enum.empty?(accept_encoding) and
```

Comment:
Yes but there will be code duplication because we have the same action to di when accept_encoding is empty and when it is not empty bu don't contain "gzip"

Reply:
You can use `with` statement to chain conditions